kernel/src/engine/default/storage.rs

```diff
@@ -1,38 +1,120 @@
-#[cfg(feature = "cloud")]
-use hdfs_native_object_store::HdfsObjectStore;
 use object_store::parse_url_opts as parse_url_opts_object_store;
 use object_store::path::Path;
 use object_store::{Error, ObjectStore};
 use url::Url;
 
+use std::collections::HashMap;
+use std::sync::{Arc, LazyLock, RwLock};
+
+/// Alias for convenience
+type ClosureReturn = Result<(Box<dyn ObjectStore>, Path), Error>;
+/// This type alias makes it easier to reference the handler closure(s)
+///
+/// It uses a HashMap<String, String> which _must_ be converted in our [parse_url_opts] because we
+/// cannot use generics in this scenario.
```

Comment:
Can we say why. Specifically I _think_ it's that `Into` is _not_ "dyn compatible", so we can't just box up the traits because you cannot write:
```rust
Box<dyn IntoIterator<Item = dyn AsRef<str>, IntoIter = dyn Into<String>>>
```